packages/smart-wallet/src/proposals/upgrade-walletFactory-proposal.js

```diff
@@ -0,0 +1,252 @@
+// @ts-check
+import { E } from '@endo/far';
+import { makeMarshal } from '@endo/marshal';
+
+console.warn('@@@ upgrade-walletFactory-proposal.js module evaluating');
+
+const { fromEntries, keys, values } = Object;
+const { Fail } = assert;
+
+// vstorage paths under published.*
+const WALLET_STORAGE_PATH_SEGMENT = 'wallet';
+const BOARD_AUX = 'boardAux';
+
+const marshalData = makeMarshal(_val => Fail`data only`);
+
+// borrow zip, allValues from @agoric/interal
+// but don't bring in all of @endo/marshal etc.
+/** @type { <X, Y>(xs: X[], ys: Y[]) => [X, Y][]} */
+const zip = (xs, ys) => harden(xs.map((x, i) => [x, ys[+i]]));
```

Comment:
I see an import from `@endo/marshal` above, so the comment seems a little off. Is there a reason why importing from `@agoric/internal` as well would be troublesome?

Reply:
I think we're no longer planning to install this bundle separate from a chain upgrade, so I think it's no longer a concern.

But if it's done with an InstallBundle tx, there's a roughly 2 cents per kilobyte charge that adds up to thousands of IST.

Reply:
can confirm: we're doing upgrade-12 as a chain upgrade so not incurring the InstallBundle expense

Better for clarity to import agoric/internal.